// Code generated by mockery v2.9.4. DO NOT EDIT.

package usecase

import mock "github.com/stretchr/testify/mock"

// MockUseCaseError is an autogenerated mock type for the UseCaseError type
type MockUseCaseError struct {
	mock.Mock
}

// Code provides a mock function with given fields:
func (_m *MockUseCaseError) Code() errorCode {
	ret := _m.Called()

	var r0 errorCode
	if rf, ok := ret.Get(0).(func() errorCode); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(errorCode)
	}

	return r0
}

// Error provides a mock function with given fields:
func (_m *MockUseCaseError) Error() string {
	ret := _m.Called()

	var r0 string
	if rf, ok := ret.Get(0).(func() string); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(string)
	}

	return r0
}
